interfaces/compiler/src/main/java/compiler/interfaces/MigrateReporter.java

```diff
@@ -0,0 +1,150 @@
+                        return Optional.empty();
+
+                    return Optional.ofNullable(src.file().path());
+                }
+                public Optional<java.io.File> sourceFile() {
+                    if (!src.exists())
+                        return Optional.empty();
+
+                    return Optional.ofNullable(src.file().file());
+                }
+                public Optional<Integer> line() {
+                    if (src.content().length == 0)
+                        return Optional.empty();
+
+                    int line = pos.line() + 1;
+                    if (line == -1)
+                        return Optional.empty();
+
+                    return Optional.of(line);
+                }
+                public String lineContent() {
+                    if (src.content().length == 0)
+                        return "";
+
+                    String line = pos.lineContent();
+                    if (line.endsWith("\r\n"))
+                        return line.substring(0, line.length() - 2);
+                    if (line.endsWith("\n") || line.endsWith("\u000c"))
+                        return line.substring(0, line.length() - 1);
+
+                    return line;
+                }
+                public Optional<Integer> offset() {
+                    return Optional.of(pos.point());
+                }
+                public Optional<Integer> pointer() {
+                    if (src.content().length == 0)
+                        return Optional.empty();
+
+                    return Optional.of(pos.point() - src.startOfLine(pos.point()));
+                }
+                public Optional<String> pointerSpace() {
+                    if (src.content().length == 0)
+                        return Optional.empty();
+
+                    String lineContent = this.lineContent();
+                    int pointer = this.pointer().get();
+                    StringBuilder result = new StringBuilder();
+                    // Don't crash if pointer is out-of-bounds (happens with some macros)
+                    int fixedPointer = Math.min(pointer, lineContent.length());
+                    for (int i = 0; i < fixedPointer; i++)
+                        result.append(lineContent.charAt(i) == '\t' ? '\t' : ' ');
+                    return Optional.of(result.toString());
+                }
+            };
+        } else {
+            position = noPosition;
+        }
+
+
+        Message message = dia.msg();
+        StringBuilder rendered = new StringBuilder();
+        rendered.append(messageAndPos(message, dia.pos(), diagnosticLevel(dia), ctx));
+
+        boolean shouldExplain = dotty.tools.dotc.reporting.Diagnostic.shouldExplain(dia, ctx);
+        if (shouldExplain && !message.explanation().isEmpty()) {
+            rendered.append(explanation(message, ctx));
+        };
+        
+        delegate.log(new MyProblem(position, message.msg(), severity, rendered.toString()));
```

Comment:
Why do you use the reporter and not the scribe logger?

```scala
dia.level match {
  case Error =>
    scribe.error(rendered.toString());
  case Warning => 
    scribe.warn(rendered.toString())
  ...
}
```

If you do so you can remove all the code about converting a Diagnostic to a Problem

Reply:
I cannot access scribe from java.